impls/python/step0_repl.py

```diff
@@ -6,17 +6,16 @@ def READ(str):
     return str
 
 # eval
-def EVAL(ast, env):
-        #print("EVAL %s" % printer._pr_str(ast))
-        return ast
+def EVAL(ast):
+    return ast
```

Comment:
So here indeed `env` is not used, but it's part of the evolution of the program (you'll see that in all impls). In step 0, 1 it's not used and then in later steps it is used (and later it becomes an `Env` object instead of a dict).

Reply:
In `process/guide.md`, the `env` parameter explicitly appears in step2.  Several implementations, including the graphs in the same subdirectory, are not consistent, probably because of copy-paste between steps.